docs/design-notes/protocols.md

```diff
@@ -93,19 +93,56 @@ protocol. Scylla supports several Scylla-only extensions to this protocol,
 described in [protocol-extensions.md](protocol-extensions.md).
 
 By default, Scylla listens to the CQL protocol on port 9042, which can be
-configured via the `native_transport_port` configuration option.
+configured via the `native_transport_port` configuration option. If set
+to 0, the OS will choose the port randomly (which also applies to every CQL
+port discussed later). To explicitly disable listening on a CQL port one
+should set its value to `~` , or `null`.
```

Comment:
Wow, why did we change this? The natural "0" used to work fine for this, why did we want to break it?
(of course my comment is about the code change, in https://github.com/scylladb/scylla/commit/29e4737a9b06875ed3ef97e816252a83529c529c, not about this documentation).

Reply:
@amoskong requested that here: https://github.com/scylladb/scylla/issues/7866
Why? I don't know - probably because 0 is a standard, valid wildcard port (?)

Reply:
I can't believe I missed this :-( Issue #7866 makes no sense to me, and it seems it brought a long chain of patches which complicated everything - almost to absurd levels (I honestly didn't understand what that matrix thing is...), for no good reason. I'm starting to wonder if I should propose to revert #7866 and what followed it :-(
In any case, I'm glad you are documenting the situation now - because it was the first time I realized what has happened here, and I wonder if I'm the only one who is living under a rock and didn't notice - or other people didn't notice either.